vm.go

```diff
@@ -3440,9 +3440,9 @@ repeat:
 		vm.popCtx()
 		vm.sp -= n + 1
 		vm.pc++
-	case *lazyObject:
-		obj.self = f.create(obj)
-		goto repeat
+	case *templatedFuncObject:
+		// todo fixes an issue
```

Comment:
We diverge a bit from upstream here, so I needed to support vmCall for templatedFuncObject types (which just meant pulling over that method) and then adding a specific case for templatedFuncObject